membership/src/test/scala/zio/membership/hyparview/plumtree/PeerStateSpec.scala

```diff
@@ -0,0 +1,127 @@
+    )(
+      i => ZIO.effectTotal(Chunk.fromArray(BigInt(i).toByteArray))
+    )
+
+    val environment =
+      (InMemoryTransport.make[Int]() ++ Logging.ignore ++ TRandom.live) >>>
+        (ZLayer.identity[Logging with TRandom with Transport[Int]] ++ TestPeerService.make(0)) >>>
+        (ZLayer
+          .identity[Transport[Int] with Logging with TRandom with TestPeerService[Int] with PeerService[Int]] ++ PeerState
+          .live(2))
+
+    suite("PeerState")(
+      testM("takes initial eager peers from PeerService") {
+        checkM(Gen.listOf(Gen.anyInt).map(_.toSet)) { peers =>
+          {
+            for {
+              _      <- TestPeerService.setPeers(peers).commit
+              result <- PeerState.live[Int](peers.size).build.use(_.get[PeerState.Service[Int]].eagerPushPeers.commit)
+            } yield assert(result)(hasSameElements(peers))
+          }.provideCustomLayer(environment)
+        }
+      },
+      testM("has lazy peers as an empty set initially") {
+        checkM(Gen.listOf(Gen.anyInt).map(_.toSet)) { peers =>
+          {
+            for {
+              _      <- TestPeerService.setPeers(peers).commit
+              result <- PeerState.live[Int](peers.size).build.use(_.get[PeerState.Service[Int]].lazyPushPeers.commit)
+            } yield assert(result)(isEmpty)
+          }.provideCustomLayer(environment)
+        }
+
+      },
+      testM("remove removes from eager peers") {
+        assertM(
+          ZSTM.atomically {
+            for {
+              _     <- PeerState.addToEagerPeers(1)
+              _     <- PeerState.removePeer(1)
+              peers <- PeerState.eagerPushPeers[Int]
+            } yield peers
+          }
+        )(isEmpty).provideCustomLayer(environment)
+      },
+      testM("remove removes from lazy peers") {
+        assertM(
+          ZSTM.atomically {
+            for {
+              _     <- PeerState.addToEagerPeers(1)
+              _     <- PeerState.moveToLazyPeers(1)
+              _     <- PeerState.removePeer(1)
+              peers <- PeerState.lazyPushPeers[Int]
+            } yield peers
+          }
+        )(isEmpty).provideCustomLayer(environment)
+      },
+      testM("addToEagerPeers adds to eager peers") {
+        checkM(Gen.listOf(Gen.anyInt).map(_.toSet)) { peers =>
+          assertM((ZSTM.foreach(peers)(PeerState.addToEagerPeers) *> PeerState.eagerPushPeers[Int]).commit)(
+            hasSameElements(peers)
+          ).provideCustomLayer(environment)
+        }
+      },
+      testM("addToEagerPeers removes from lazy peers") {
+        assertM(
+          ZSTM.atomically {
+            for {
+              _     <- PeerState.addToEagerPeers(1)
+              _     <- PeerState.moveToLazyPeers(1)
+              _     <- PeerState.addToEagerPeers(1)
+              peers <- PeerState.lazyPushPeers[Int]
+            } yield peers
+          }
+        )(isEmpty).provideCustomLayer(environment)
+      }
+      // TODO: move to PlumTree tests
```

Comment:
can you clean this up now?

Reply:
I would, but there is no test suite for PlumTree yet. I'm still thinking about how to best do that without going insane in the process.

I'll take care of them before tackling any new major features